### PR TITLE
Add support for all Discord languages and translation system.

### PR DESCRIPTION
**Pull request by LautyDev**
Hello, good afternoon, we want to tell you that together with [tnfAngel](https://github.com/tnfAngel) we have developed a translation system and we have edited it to support YML since the original version (https://github.com/FNLB-Project/Perception) uses JSON. We have added **all Discord languages** but you have to take into account the following:

- It is possible that in the translations spaces are added that are unnecessary because that's how Google translator works, you will have to fix them by hand.
- We recommend you not to put the emojis (ex: :no:) in the YML, but like this: {{emoji}} to avoid problems.

The only thing we ask for is **acknowledgement**, either on the Github, on the server or on the bot. I hope you like the system and have a good start of the year.

jjLpmXX 1

### CODE REVIEW

**Comment by browniepy** [comment]
https://github.com/Discord-InterChat/locales/assets/68281832/eab8e954-ad68-412a-b121-81bd1934968b

**Comment by browniepy** [comment]
🔒💿👖💃👙🦀🛌😎🗣️🔥👋😘👃🚌🤙📈✅😋❌😭☕💬👻😈🍌😺🍫🎲🦊🍓🐢❤️💤🌝🌛🌜

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+const wait = (time: number) => new Promise((res) => setTimeout(res, time));
+
+const getArrayKey = (array: string | string[]) =>
+	Array.isArray(array) ? array[0] : array;
+
+const getArrayValue = (array: string | string[]) =>
+	Array.isArray(array) ? array[1] : array;
+
+const writeYamlFileSync = (filePath: string, content: Record<string, any>) => {
+	const yamlContent = yaml.dump(content);
```
Comment (dev-737) [inline src/index.ts]:
```suggestion
	const yamlContent = yaml.dump(content, { lineWidth: Number.MAX_SAFE_INTEGER });
```
Its better to remove these new lines created by line width limits. Makes embeds look bulky

**Comment by dev-737** [comment]
I've looked through the code, looks good. 🙂 Just waiting for the language experts to look at the translations. Let's wait for their thoughts before proceeding.

**Comment by dev-737** [comment]
Update on this, we had around 3 different language speakers look at this so far, it looks like that the translations provided by Google translate were inaccurate and hard to understand in some places. Would it be possible to utilise some other more-accurate translation API for this?

**Comment by tnfAngel** [comment]
Any recommendation about a free and good API? By the way, automatically translated languages ​​are intended to be a "base" as they can be modified and will not re-translate what was modified, so it is a good base and can be modified.

**Comment by cloudcloudskies** [comment]
No, it's a bad base. Syntax's are messed up. Translations are so bad. Which will take a long time to fix.

**Comment by tnfAngel** [comment]
> No, it's a bad base. Syntax's are messed up. Translations are so bad. Which will take a long time to fix.

Then we invite you to translate it into all Discord languages by ​​yourself ☺️

**Comment by LautyDev** [comment]
@lunar-net Good morning, again we ask you (it is already the fourth time) to have a little respect and if you are not happy with the translations you can run the system on your own and make the improvements that you think are right (if you do, we would like to see them). On the other hand the syntax issue is something that was already warned in the Discord, we recommend you not to pass the emojis and links through the YML file (besides that it is code repetition). I will use as an example my mother tongue (Spanish), I have read some text and personally I understand perfectly what it means but as we have said it is a base and it can be edited perfectly. Again remember respect and clean dialogue, we are only here to help. Best regards.

**Comment by dev-737** [comment]
> Any recommendation about a free and good API? By the way, automatically translated languages ​​are intended to be a "base" as they can be modified and will not re-translate what was modified, so it is a good base and can be modified.

Afaik DeepL is a good alternative, it can be used both as a base and also as-is. The only problem is obtaining an API key, which is fortunately free, but requires payment information nonetheless :/

Further, let's get input from speakers of other languages to get a feel for which languages Google translate is good at, and use them as base translations. Meanwhile, I would greatly appreciate it if you could get input from other language speakers as well and let me know on discord :)

File: src/locales/tr.yml [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+
+   Desteğiniz için minnettarlık ifade etme yolumuz olarak, size özel avantajlar sunmaktan heyecan duyuyoruz. İnterchat için oylamayı yayınlayarak kilidini açacaksınız: 
+
+   - Hub'lar içindeki mesajları düzenleyin 
+   - Çeviri mesajları (ve çok daha fazlası yolda!) 
+
+   Değişmez desteğinizi derinden takdir ediyoruz. Teşekkür ederim! 🙏
+network:
+  deleteSuccess: >-
+    {{{emoji}}}}} {{{user}}} tarafından mesaj __ ** {{deleted}}/{{total}}} ** __
```
Comment (dev-737) [inline src/locales/tr.yml]:
```suggestion
    {{{emoji}}} {{{user}}} tarafından mesaj __ ** {{deleted}}/{{total}}} ** __
```
Looks like extra curly brackets are getting added for tr, cs, el, fi, ja, ko, lt, nl, no, th and zh translations. Could you please check? 🤔

**Review by dev-737 — CHANGES_REQUESTED** [review]
@LautyDev After confirming with some translators of other languages, we can proceed with adding this! ~~Please resolve the conflicts and check the reviews so we can proceed~~ (Found out it was possible for me to commit to forks 💀 I fixed all the problems except for the emojis. I'll check about that later.)

**Review by dev-737 — APPROVED** [review]

**Pull request merged**